dor/providers/process_basic_image.py

```diff
@@ -114,7 +244,70 @@ def get_service_file_info(file_set_identifier: FileSetIdentifier) -> FileInfo:
         identifier=file_set_identifier.identifier,
         basename=file_set_identifier.basename,
         uses=[UseFunction.service, UseFormat.image],
-        mimetype=ImageMimetype.JP2.value
+        mimetype=ImageMimetype.JP2.value,
+    )
+
+
+def process_source_file(image_path: Path, transformer: Transformer, uses: list[Any]):
+    try:
+        source_tech_metadata = ImageTechnicalMetadata.create(image_path)
+    except JHOVEDocError:
+        return None
```

Comment:
@botimer the transformer isn't doing anything with errors. I think you had ideas of collecting errors vs. exiting early? I was interpreting that as needing something like `transformer.log_error(exception)`.

Reply:
I am having flashbacks:

- https://github.com/mlibrary/moku/blob/master/lib/moku/command/deploy.rb
- https://github.com/mlibrary/moku/blob/master/lib/moku/pipeline/deploy.rb#L32-L41
- https://github.com/mlibrary/moku/blob/master/lib/moku/pipeline/pipeline.rb#L45-L55
- https://github.com/mlibrary/moku/blob/master/lib/moku/plan/basic_deploy.rb#L21-L41
- https://github.com/mlibrary/moku/blob/master/lib/moku/plan/plan.rb#L32-L38
- https://github.com/mlibrary/moku/blob/master/lib/moku/task/remote_shell.rb#L25-L31
- https://github.com/mlibrary/moku/blob/master/lib/moku/shell/passthrough.rb#L33-L43

I need to process more to make meaningful commentary about Transformer. _I've been here before._

Reply:
_Looks at some links_. Oh geez, sorry about that. And this isn't even the version that had a `Command` class. 😬